internal/providers/providers.go

```diff
@@ -212,39 +216,19 @@ func convertRecordToEndpoint(record records.RecordList, domainName string) *endp
 	if record.Type == "NS" || record.Type == "SOA" {
 		return nil
 	}
-
-	domainName = strings.TrimSuffix(strings.ToLower(domainName), ".")
-	recordName := strings.TrimSuffix(strings.ToLower(record.Name), ".")
-
-	var dnsName string
-	if recordName == "" || recordName == domainName {
-		dnsName = domainName + "."
-	} else if strings.HasSuffix(recordName, "."+domainName) {
-		dnsName = recordName + "."
-	} else {
-		dnsName = recordName + "." + domainName + "."
-	}
-
-	// Normalize TXT data (Rackspace often stores without quotes)
-	data := record.Data
-	if record.Type == "TXT" {
-		data = strings.Trim(data, `"`)
-		data = fmt.Sprintf(`"%s"`, data)
```

Comment:
These entries are being removed because when external-dns sends requests to `POST /adjustendpoint`
this code causes a mismatch between what is stored in Cloud DNS and what exists in the cluster.

As a result, external-dns assumes that the Cloud DNS entries have changed, so it deletes them and then attempts to recreate/update them.